### PR TITLE
Create beta build

### PR DESCRIPTION
**Pull request by paullinator**
#### PR Dependencies

 none

#### PR Requirements

If you have made **any** visual changes to the GUI. Make sure you have:
- [x] Tested on iOS device
- [x] Tested on Android device
- [x] Tested on small-screen device (iPod Touch)
- [ ] Tested on large-screen device (tablet)

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203053557504865

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -49,6 +49,15 @@ pipeline {
...
         }
 
+        // develop credentials
+        withCredentials([
+          file(credentialsId: "efbd6cab-a1b3-4027-9a15-8c76fe7c1350", variable: "GoogleService_Info"),
+          file(credentialsId: "36c06912-38a6-466c-98dd-9ddd98ec7fd4", variable: "google_services"),
+        ]) {
+          sh "cp ${GoogleService_Info} ./deployPatches/edge/develop/GoogleService-Info.plist"
+          sh "cp ${google_services} ./deployPatches/edge/develop/google-services.json"
+        }
```
Comment (samholmes) [inline Jenkinsfile:61]: Why are separate `GoogleService-Info.plist` and `google-services.json` files needed for beta and develop builds anyway?

File: scripts/deploy.ts [hunk 2]
```diff
@@ -98,6 +100,7 @@ function makeCommonPre(argv: string[], buildObj: BuildObj) {
   buildObj.projectName = argv[2]
   buildObj.guiPlatformDir = buildObj.guiDir + '/' + buildObj.platformType
   buildObj.tmpDir = `${buildObj.guiDir}/temp`
+  buildObj.buildArchivesDir = '/Users/jenkins/buildArchives'
```
Comment (samholmes) [inline scripts/deploy.ts:103]: Shouldn't we get the user dir from the environment. It'd be a rare case, so probably not worthwhile, but could the build server username change from `jenkins`?

File: Jenkinsfile [hunk 3]
```diff
@@ -53,18 +53,22 @@ pipeline {
         withCredentials([
           file(credentialsId: "efbd6cab-a1b3-4027-9a15-8c76fe7c1350", variable: "GoogleService_Info"),
           file(credentialsId: "36c06912-38a6-466c-98dd-9ddd98ec7fd4", variable: "google_services"),
+          file(credentialsId: "5bbb9f5b-fa33-4f76-9796-0e0638840fe6", variable: "edge_develop_keystore"),
         ]) {
           sh "cp ${GoogleService_Info} ./deployPatches/edge/develop/GoogleService-Info.plist"
           sh "cp ${google_services} ./deployPatches/edge/develop/google-services.json"
+          sh "cp ${edge_develop_keystore} ./keystores/edge-develop-keystore.jks"
```
Comment (samholmes) [inline Jenkinsfile:60]: What is this file for?

File: scripts/createAndroidKeys.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    killSignal: 'SIGKILL'
+  })
+}
+
+function chdir(path: string) {
+  console.log('chdir: ' + path)
+  _currentPath = path
+}
+
+main().catch(e => console.log(e.message))
```
Comment (samholmes) [inline scripts/createAndroidKeys.ts:71]: console.error here too?

File: scripts/createAndroidKeys.ts [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+import prompts from 'prompts'
+
+let _currentPath = __dirname
+
+const main = async (): Promise<void> => {
+  // let alias = ''
+
+  mylog(_currentPath)
+  const newpath = join(_currentPath, '..', 'keystores')
+  mylog(newpath)
```
Comment (samholmes) [inline scripts/createAndroidKeys.ts:14]: chdir already logs the `newpath` it seems

File: scripts/createAndroidKeys.ts [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+  //   `keytool -export -rfc -keystore ${alias}-upload-keystore.jks -alias ${alias}-upload -file ${alias}-upload-cert.pem -storepass ${password} -keypass ${password}`
+  // )
+
+  mylog('Keystore created in keystores directory. Save the following keystore password. It cannot be recovered if lost.')
+  mylog('********************************')
+  mylog(password)
+  mylog('********************************')
+}
+
+const mylog = console.log
```
Comment (samholmes) [inline scripts/createAndroidKeys.ts:53]: I'd argue to move this constant to the top of the file

**Review by samholmes — CHANGES_REQUESTED** [review]
Minor feedback and which may warrant changes after discussion.

File: scripts/createAndroidKeys.ts (hunk 6 above)
Comment (paullinator) [reply scripts/createAndroidKeys.ts:53]: Per prior comments from William, utility function should be at the end of the file so you can see the primary routines at the top.

Comment (samholmes) [reply scripts/createAndroidKeys.ts:50]: Ah, fair.

**Review by samholmes — APPROVED** [review]

**Pull request merged**